src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerDependencyCheckingService.cs

```diff
@@ -19,79 +18,131 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation
+            _hostDiagnosticUpdateSource = hostDiagnosticUpdateSource;
         }
 
-        public async void CheckForConflictsAsync()
+        public void ReanalyzeSolutionForConflicts()
         {
-            AnalyzerDependencyResults results = null;
-            try
-            {
-                results = await GetConflictsAsync().ConfigureAwait(continueOnCapturedContext: true);
-            }
-            catch
-            {
-                return;
-            }
+            var solution = _workspace.CurrentSolution;
+            var currentAnalyzerPaths = solution
+                .Projects
+                .SelectMany(p => p.AnalyzerReferences)
+                .OfType<AnalyzerFileReference>()
+                .Select(a => a.FullPath)
+                .ToImmutableHashSet(StringComparer.OrdinalIgnoreCase);
 
-            if (results == null)
+            lock (_gate)
             {
-                return;
+                // If we've already started analysis for this set of analyzers, no reason to start over
+                if (currentAnalyzerPaths.SetEquals(_previousAnalyzerPaths))
+                {
+                    return;
+                }
+
+                _cancellationTokenSource.Cancel();
+                _cancellationTokenSource = new CancellationTokenSource();
+                _previousAnalyzerPaths = currentAnalyzerPaths;
+
+                // We are explicitly relying on SafeContinueWith including LazyCancellation as a continuation option here
+                _task = _task.SafeContinueWith(_ =>
+                {
+                    AnalyzeAndReportConflictsInSolution(solution, currentAnalyzerPaths, _hostDiagnosticUpdateSource, _cancellationTokenSource.Token);
+                },
+                _cancellationTokenSource.Token, TaskScheduler.Default);
             }
+        }
+
+        // Method is static to prevent accidental use of mutable state in this class
+        private static void AnalyzeAndReportConflictsInSolution(
+            Solution solution,
+            IEnumerable<string> currentAnalyzerPaths,
+            HostDiagnosticUpdateSource hostDiagnosticUpdateSource,
+            CancellationToken cancellationToken)
+        {
+            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies().Select(assembly => AssemblyIdentity.FromAssemblyDefinition(assembly));
+            var loadedAssembliesList = new IgnorableAssemblyIdentityList(loadedAssemblies);
+
+            var ignorableAssemblyLists = new[] { s_systemPrefixList, s_codeAnalysisPrefixList, s_explicitlyIgnoredAssemblyList, s_assembliesIgnoredByNameList, loadedAssembliesList };
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var results = AnalyzerDependencyChecker.ComputeDependencyConflicts(currentAnalyzerPaths, ignorableAssemblyLists, s_bindingRedirectionService, cancellationToken);
 
             var builder = ImmutableArray.CreateBuilder<DiagnosticData>();
```

Comment:
arraypool?

Reply:
Cost would be tiny compared to the allocations happening earlier...